app/models/miq_request_workflow.rb

```diff
@@ -47,24 +47,30 @@ def self.update_requester_from_parameters(data, user)
     new_user
   end
 
-  def initialize(values, requester, options = {})
-    instance_var_init(values, requester, options)
+  def initialize_dialogs(values, options)
+    return if options[:skip_dialog_load] == true
 
-    unless options[:skip_dialog_load] == true
-      # If this is the first time we are called the values hash will be empty
-      # Also skip if we are being called from a web-service
-      if @dialogs.nil?
-        @dialogs = get_dialogs
-        normalize_numeric_fields
-      else
-        @running_pre_dialog = true if options[:use_pre_dialog] != false
-      end
+    # If this is the first time we are called the values hash will be empty
+    # Also skip if we are being called from a web-service
+    if @dialogs.nil?
+      @dialogs = get_dialogs
+    else
+      @running_pre_dialog = true if options[:use_pre_dialog] != false
     end
+    normalize_numeric_fields unless @dialogs.nil?
```

Comment:
I'm curious why this was move from the `if @dialogs.nil?` branch

Reply:
that is the way some of the subclasses had it.  so I commonized the code.